test/Network/Socket.test.cpp

```diff
@@ -1,9 +1,114 @@
 #include <SFML/Network/Socket.hpp>
 
+#include <catch2/catch_test_macros.hpp>
+
 #include <type_traits>
 
-static_assert(!std::is_constructible_v<sf::Socket>);
-static_assert(!std::is_copy_constructible_v<sf::Socket>);
-static_assert(!std::is_copy_assignable_v<sf::Socket>);
-static_assert(!std::is_nothrow_move_constructible_v<sf::Socket>);
-static_assert(!std::is_nothrow_move_assignable_v<sf::Socket>);
+class TestSocket : public sf::Socket
+{
+public:
+    TestSocket() : sf::Socket(sf::Socket::Type::Udp)
+    {
+    }
+
+    sf::SocketHandle getHandle() const
+    {
+        return sf::Socket::getHandle();
+    }
```

Comment:
This could be `using Socket::getHandle;`
Same for methods `create` and `close`.
This would make both `Socket::create()` and `Socket::create(SocketHandle)` accessible even though we directly call only `Socket::create()`, but I don't think that would be a problem.

Reply:
Ooooh I love this trick